src/effects.ts

```diff
@@ -8,79 +10,230 @@ import 'rxjs/add/operator/catch';
+    private jsonApi: NgrxJsonApi,
+    private store: Store<any>,
+  ) { }
 
+  @Effect() createResource$ = this.actions$
+    .ofType(NgrxJsonApiActionTypes.API_CREATE_INIT)
+    .map<Action, Payload>(toPayload)
+    .mergeMap((payload: Payload) => {
+      return this.jsonApi.create(payload)
+        .mapTo(new ApiCreateSuccessAction(payload))
+        .catch(() => Observable.of(
+          new ApiCreateFailAction(payload)
+        ));
+    });
+
+  @Effect() updateResource$ = this.actions$
+    .ofType(NgrxJsonApiActionTypes.API_UPDATE_INIT)
+    .map<Action, Payload>(toPayload)
+    .mergeMap((payload: Payload) => {
+      return this.jsonApi.update(payload)
+        .mapTo(new ApiUpdateSuccessAction(payload))
+        .catch(() => Observable.of(new ApiUpdateFailAction(payload)));
+    });
+
+  @Effect() readResource$ = this.actions$
+    .ofType(NgrxJsonApiActionTypes.API_READ_INIT)
+    .map<Action, Payload>(toPayload)
+    .mergeMap((payload: Payload) => {
+      return this.jsonApi.find(payload)
+        .map(res => res.json())
+        .map(data => new ApiReadSuccessAction({
+          jsonApiData: data,
+          query: payload.query
+        }))
+        .catch(() => Observable.of(new ApiReadFailAction(payload)));
+    });
+
+  @Effect() deleteResource$ = this.actions$
+    .ofType(NgrxJsonApiActionTypes.API_DELETE_INIT)
+    .map<Action, Payload>(toPayload)
+    .mergeMap((payload: Payload) => {
+      return this.jsonApi.delete(payload)
+        .mapTo(new ApiDeleteSuccessAction(payload))
+        .catch(() => Observable.of(new ApiDeleteFailAction(payload)));
+    });
+
+  @Effect() commitResources$ = this.actions$
+    .ofType(NgrxJsonApiActionTypes.API_COMMIT_INIT)
+    .map<Action, string>(toPayload)
+    .mergeMap((storeLocation : string) => this.store.select(storeLocation).take(1))
+    .mergeMap((store : NgrxJsonApiStore) => {
+      // TODO add support for bulk updates as well (jsonpatch, etc.)
+      // to get atomicity for multiple updates
+
+      console.log("commit", store);
+
+      let pending : Array<StoreResource> = this.getPendingChanges(store);
+      console.log("pending", pending);
+
+      if(pending.length > 0){
+        pending = this.sortPendingChanges(pending);
+
+        let actions : Array<Observable<Action>> = [];
+        for(let pendingChange of pending){
+          if(pendingChange.state == ResourceState.CREATED){
+            let payload : Payload = {
+              jsonApiData : {
+                data : {
+                  id :  pendingChange.resource.id,
+                  type :  pendingChange.resource.type,
+                  attributes :  pendingChange.resource.attributes,
+                  relationships :  pendingChange.resource.relationships
+                },
+              },
+              query : {
+                queryType : 'create',
+                type : pendingChange.resource.type
+              }
+            };
+            actions.push(this.jsonApi.create(payload)
```

Comment:
Why not utilise `API_CREATE_INIT` and update that action so it handles errors like you did here.